### ISSUE TITLE
Code for Continual learning applications

### ISSUE THREAD

**Issue by thomascong121** [VICO-UoE/DatasetCondensation#28]
Thanks for the great works, I would like to ask where can we find the code for task incremental learning applications conducted in your first two papers (SVHN→MNIST→USPS). It seems like the current CL_DM.py only works for class incremental learning setups in Distribution Matching paper.

**Comment by PatrickZH** [comment]
Sorry, we didn't provide the code for SVHN->MNIST->USPS. The reasons are 1) CL needs specific splitting of training and testing protocol for different datasets, 2) the new setting is more challenging and practical than the previous one.

**Comment by thomascong121** [comment]
Can I get some more details about this setup? for SVHN->MNIST->USPS training, you first init a model A and train it on SVHN training set and test on SVHN test set, then you train on MNIST training set and SVHN syn set and test on both test sets and so on so forth, is this the right setting? also the GDumb method used in Distribution Matching paper should also work on this task incremental learning setup, right?

**Comment by PatrickZH** [comment]
Actually, when moving to new dataset, we condense the new dataset into a synthetic one, and then continuously train the model on the joint synthetic datasets. In GDumb, they first do coreset selection on new dataset and then train models on the joint coresets.

**Comment by thomascong121** [comment]
Alright, thanks for the explanation, I will give it a go. I do have another question, I read through your papers, but I probably missed your explanation of choosing ResNetAp-10 for distilled sample training, can you please give me some reasons for this?

**Comment by PatrickZH** [comment]
Please refer to the bottom paragraph in page 13 of https://arxiv.org/pdf/2006.05929.pdf , which starts with "In the first stage – while training the condensed image..."

**Comment by thomascong121** [comment]
Thanks a lot.

**Comment by thomascong121** [comment]
Hi Bo, for dataset distillation, do you think it is possible to jointly optimize the downstream task with the data distillation module, so that the synthetic data is more fit to the downstream tasks?

**Comment by BAAI-DCAI** [comment]
The downstream tasks have been optimized as mimic tasks in the innerloop of dataset distillation.

**Comment by thomascong121** [comment]
> The downstream tasks have been optimized as mimic tasks in the innerloop of dataset distillation.

Yes, I understand that. I just wander if it is possible to update the synthetic data and the model at the same time. In other words, instead of updating $\theta_{t+1}$ on line 9 in alg 1, can we update $\theta_{t+1}$ together with $\mathcal{S}_{c}$ on line 8?

**Comment by BAAI-DCAI** [comment]
Feel free to have a try.
